extra/scanner.xsd

```diff
@@ -0,0 +1,97 @@
+    </xs:simpleType>
+    <xs:element name="document">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element name="scanner" maxOccurs="unbounded" minOccurs="1">
+                    <xs:complexType mixed="true">
+                        <xs:sequence>
+                            <xs:element name="FWFSettings" minOccurs="0">
+                                <xs:complexType>
+                                    <xs:simpleContent>
+                                        <xs:extension base="xs:string">
+                                            <xs:attribute type="xs:int" name="beamSampleQuality"/>
+                                            <xs:attribute type="xs:float" name="binSize_ns"/>
+                                            <xs:attribute type="xs:float" name="maxFullwaveRange_ns"/>
+                                            <xs:attribute type="xs:float" name="winSize_ns"/>
+                                        </xs:extension>
+                                    </xs:simpleContent>
+                                </xs:complexType>
+                            </xs:element>
+                            <xs:element name="beamOrigin" minOccurs="0">
+                                <xs:complexType>
+                                    <xs:choice maxOccurs="unbounded" minOccurs="0">
+                                        <xs:element name="rot">
+                                            <xs:complexType>
+                                                <xs:simpleContent>
+                                                    <xs:extension base="xs:string">
+                                                        <xs:attribute type="axisType" name="axis" use="optional"/>
+                                                        <xs:attribute type="xs:float" name="angle_deg" use="optional"/>
+                                                    </xs:extension>
+                                                </xs:simpleContent>
+                                            </xs:complexType>
+                                        </xs:element>
+                                    </xs:choice>
+                                    <xs:attribute type="xs:float" name="x" use="optional"/>
+                                    <xs:attribute type="xs:float" name="y" use="optional"/>
+                                    <xs:attribute type="xs:float" name="z" use="optional"/>
+                                    <xs:attribute type="rotType" name="rotations" use="optional"/>
+                                </xs:complexType>
+                            </xs:element>
+                            <xs:element name="headRotateAxis" minOccurs="0">
+                                <xs:complexType>
+                                    <xs:simpleContent>
+                                        <xs:extension base="xs:string">
+                                            <xs:attribute type="xs:byte" name="x" use="optional"/>
```

Comment:
Ah, this was due to the xsd generator that I used, which automatically determined this type because in the sample XML, x, y and z all either had 0 or 1. But I guess we can also specify floats, so that the head rotates around another axis than one of the three coordinate axes, is this correct? I'll change it to float then

Reply:
yes, one could imagine an axis tilted by 45 deg, then it would be sqrt(2)/2, sqrt(2)/2, 0